packages/frontend/src/hooks/useEstimateTxCost.ts

```diff
@@ -81,62 +94,67 @@ export function useEstimateTxCost(selectedNetwork?: Network) {
           destinationDeadline = 0
         }
 
-        // Get estimated gas limit
-        const estimatedGasLimit = await bridge.estimateSendGasLimit(
-          '10',
-          fromNetwork.slug as string,
-          toNetwork.slug as string,
-          {
-            recipient: constants.AddressZero,
-            bonderFee: '1',
-            amountOutMin: '0',
-            deadline: deadline(),
-            destinationAmountOutMin,
-            destinationDeadline,
-          }
+        const needsAppoval = await token.needsApproval(
+          bridge.getSendApprovalAddress(fromNetwork.slug),
+          '10'
         )
 
+        let estimatedGasLimit = BigNumber.from(200e3)
+        // Get estimated gas limit
+        if (!needsAppoval) {
+          estimatedGasLimit = await bridge.estimateSendGasLimit(
+            '10',
+            fromNetwork.slug as string,
+            toNetwork.slug as string,
+            {
+              recipient: constants.AddressZero,
+              bonderFee: '1',
+              amountOutMin: '0',
+              deadline: deadline(),
+              destinationAmountOutMin,
+              destinationDeadline,
+            }
+          )
+        }
+
         if (estimatedGasLimit) {
-          let gasCost = await estimateGasCost(fromNetwork, estimatedGasLimit)
+          let gasCost = await getGasCostByGasLimit(fromNetwork.provider, estimatedGasLimit)
           if (gasCost && fromNetwork.slug === ChainSlug.Optimism) {
             const l1FeeInWei = await bridge.getOptimismL1Fee(fromNetwork.slug, toNetwork.slug)
             gasCost = gasCost.add(l1FeeInWei)
           }
           return gasCost
         }
-      } catch (error) {
-        logger.error(error)
+      } catch (error: any) {
+        logger.error(formatError(error))
       }
     },
-    [sdk, selectedNetwork]
+    [sdk]
   )
 
-  const estimateWrap = useCallback(
-    async (options: { token: Token; network: Network }) => {
-      const { token, network } = options
-      if (!network) {
-        return
-      }
-
-      try {
-        // Get estimated gas cost
-        const estimatedGasLimit = await token.wrapToken(BigNumber.from(10), true)
-
-        if (BigNumber.isBigNumber(estimatedGasLimit)) {
-          let gasCost = await estimateGasCost(network, estimatedGasLimit)
-          if (gasCost && network.slug === ChainSlug.Optimism) {
-            const { gasLimit, data, to } = await token.getWrapTokenEstimatedGas(network.slug)
-            const l1FeeInWei = await token.estimateOptimismL1FeeFromData(gasLimit, data, to)
-            gasCost = gasCost.add(l1FeeInWei)
-          }
-          return gasCost
+  const estimateWrap = useCallback(async (options: EstimateTxOptions) => {
+    const { token, network } = options
+    if (!network) {
+      return
+    }
+
+    try {
+      // Get estimated gas cost
+      const estimatedGasLimit = await token.wrapToken(BigNumber.from(10), true)
```

Comment:
suggestion: the `true` parameter should be a variable for clarity: `const estimateOnly = true`

Reply:
i did not realize we had a method `getWrapTokenEstimatedGas` for this already